tests/integration/it_poster_test.py

```diff
@@ -1,15 +1,24 @@
-import pytest
+from unittest import mock
 
-from timeless.poster.api import Poster, Authenticated
+from timeless.poster.api import Authenticated, PosterAuthData
 
 """Integration tests for Poster"""
 
 
-@pytest.mark.skip(
-    reason="Authentication mechanism (Authenticated#auth()) is not yet implemented!"
-)
 def test_auth():
-    assert Authenticated(
-        clientid="$0m3C1i3ntId"
-    ).access_token(), "Poster did not authenticate the user!"
+    auth_data = PosterAuthData(
+        application_id="test_id",
+        application_secret="test_secret",
+        redirect_uri="test_uri",
+        code="test_code",
+    )
+    auth_token = "test_auth_token"
 
+    def auth(*args, **kwargs):
+        mocked = mock.Mock()
+        mocked.return_value = auth_token
+        return mocked
+
+    with mock.patch.object(Authenticated, 'auth', auth):
```

Comment:
@vladarefiev It thinks it's a bit artificial test. Let's mock response from server. WDYT?

Reply:
@vryazanov Oops, you are stealing money from rultor :)